src/controllers/commands/CommandController.cpp

```diff
@@ -889,9 +895,27 @@ void CommandController::initialize(Settings &, Paths &paths)
                 return "";
             }
 
-            channel->addMessage(makeSystemMessage(
-                QString("Chatter count: %1")
-                    .arg(localizeNumbers(twitchChannel->chatterCount()))));
+            auto addChatterCountMessage = [channel, twitchChannel]() -> void {
+                channel->addMessage(makeSystemMessage(
+                    QString("Chatter count: %1")
+                        .arg(localizeNumbers(twitchChannel->chatterCount()))));
+            };
+
+            // Fetch from chatter cache when user does not have mod rights
+            if (!twitchChannel->hasModRights()) {
+                addChatterCountMessage();
+                return "";
+            }
+
+            // Use helix api when user has mod rights
+            twitchChannel->refreshChatters(
```

Comment:
This will end up making N requests fetching all viewers, it will probably be best to also provide a non-paginated version that this command can use directly

The way I envision it working is the user types `/chatters`, we make a request using `getHelix()->getChatters(channelsomething, /* how many users to fetch */ 1, /* pagination cursor */ boost::none, successCB, failCB)` where successCB only reads the total field and discards the rest of the payload

Reply:
Good point, I forgot there is a total in the first returned request. I'll make a separate function to just get the total. The `Helix::GetChatters` function is still relevant because we need it for refreshing the @ list.